### PR TITLE
Add MADIS package

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
Follows more or less the same formula as the hpc-stack build script. Builds successfully but not extensively tested. I have not validated whether pnetcdf is actually getting used (outside of visually inspecting the build output).

### CODE REVIEW

File: var/spack/repos/builtin/packages/madis/package.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Madis(MakefilePackage):
+    """MADIS: The Meteorological Assimilation Data Ingest System (MADIS) is
+    dedicated toward making value-added data available from the National
+    Oceanic and Atmospheric Administration's (NOAA) Earth System Research
+    Laboratory (ESRL) Global Systems Division (GSD) (formerly the Forecast
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/madis/package.py]: This affiliation is no longer correct. The Global Systems Laboratory (GSL) is now its own laboratory, directly under OAR, see https://gsl.noaa.gov/

Comment (climbfuji) [reply var/spack/repos/builtin/packages/madis/package.py]: I'd also remove the `(formerly the ... (FSL))` part

**Review by climbfuji — COMMENTED** [review]
I haven't tested this package myself. To which env is it going to be added? Can that part (adding it to an env and installing it as part of unified-dev) be done after the upcoming release?

**Comment by AlexanderRichert-NOAA** [comment]
> I haven't tested this package myself. To which env is it going to be added? Can that part (adding it to an env and installing it as part of unified-dev) be done after the upcoming release?

Yes, it's not urgent to get it into an environment, I'm adding this to make it eventually available for WCOSS2.

File: var/spack/repos/builtin/packages/madis/package.py (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/madis/package.py]: Okay, I copied and pasted from a different and more up to date website :)

**Review by climbfuji — APPROVED** [review]
Let's merge after the spack-stack 1.3.0 release. I'll add it to the next project.

**Comment by AlexanderRichert-NOAA** [comment]
Consolidating with rest of WCOSS2-related package changes.

**Pull request closed without merging**